### PR TITLE
COM-1630: map slug renaming to pageid

### PR DESCRIPTION
**Pull request by Williamas47**
Handling errors preventing the maps to load

### CODE REVIEW

**Comment by Williamas47** [comment]
There was some linting changes when I formatted the document but I'll point the main changes on `cache.js`

File: mod/workspace/cache.js [hunk 1]
```diff
@@ -81,38 +81,50 @@ async function cache() {
...
-  const { mongoClient } = req;
-  if (!pageId) {
-    return new Error('No pageId specified');
-  }
-  if (!project) {
-    return new Error('No project specified');
-  }
-  if (!lang) {
-    return new Error('No lang specified');
+  try {
```
Comment (Williamas47) [inline mod/workspace/cache.js:89]: added a try catch here

File: mod/workspace/cache.js [hunk 2]
```diff
@@ -81,38 +81,50 @@ async function cache() {
...
+    if (!project) {
+      return new Error("No project specified");
+    }
+    if (!lang) {
+      return new Error("No lang specified");
+    }
+    const { ObjectId } = require("mongodb");
+    const Page = mongoClient.db("acorn").collection("pages");
+    const Version = mongoClient.db("acorn").collection("versions");
+    const proposalPage = await Page.findOne({ _id: ObjectId(pageId) });
```
Comment (Williamas47) [inline mod/workspace/cache.js:105]: Added the `ObjectId()` into `pageId` const <- this was the main issue!

File: mod/workspace/cache.js [hunk 3]
```diff
@@ -81,38 +81,50 @@ async function cache() {
...
+    if (!proposalPage) {
+      console.log('Proposal Page not found', pageId);
```
Comment (Williamas47) [inline mod/workspace/cache.js:107]: Added some logs to debug (it was very confusing to find the origin on the error)

File: mod/workspace/cache.js [hunk 4]
```diff
@@ -81,38 +81,50 @@ async function cache() {
...
+      return;
+    }    // get the most recent version id
+    const contentVersionId = proposalPage?.content[lang][0];
+    if (!contentVersionId) {
+      console.log('Content Version Id not found');
```
Comment (Williamas47) [inline mod/workspace/cache.js:112]: more logs

File: mod/workspace/cache.js [hunk 5]
```diff
@@ -81,38 +81,50 @@ async function cache() {
...
+    }
+    const proposalContent = await Version.findOne({
+      _id: ObjectId(contentVersionId),
+    });
+    workspace = proposalContent.content.geolytixWorkspace;
+    await assignTemplates(workspace);
+
+    await assignDefaults(workspace);
+    if(!workspace){
+      console.log('No workspace found on cache.js');
```
Comment (Williamas47) [inline mod/workspace/cache.js:123]: more logs

File: mod/workspace/cache.js (hunk 2 above)
Comment (fizzy-drinks) [reply mod/workspace/cache.js:105]: Nice catch!

File: mod/workspace/cache.js (hunk 1 above)
Comment (fizzy-drinks) [reply mod/workspace/cache.js:89]: Not sure we want to obfuscate all errors here? Maybe call Sentry in the catch block?

File: mod/workspace/cache.js [hunk 6]
```diff
@@ -1,36 +1,36 @@
...
-const assignDefaults = require('./assignDefaults');
+const assignDefaults = require("./assignDefaults");
 
 let workspace = null;
 
-const { nanoid } = require('nanoid');
+const { nanoid } = require("nanoid");
 
-const logger = require('../logger');
+const logger = require("../logger");
```
Comment (fizzy-drinks) [inline mod/workspace/cache.js:15]: Not mandatory I guess but can we put all require statements at the top of the file?

File: mod/workspace/cache.js [hunk 7]
```diff
@@ -1,36 +1,36 @@
...
-const assignTemplates = require('./assignTemplates')
+const assignTemplates = require("./assignTemplates");
 
-const defaults = require('./defaults');
+const defaults = require("./defaults");
 
-const assignDefaults = require('./assignDefaults');
+const assignDefaults = require("./assignDefaults");
 
 let workspace = null;
```
Comment (fizzy-drinks) [inline mod/workspace/cache.js]: Do we really need to declare `workspace` outside the returned function?

**Review by fizzy-drinks — CHANGES_REQUESTED** [review]
Just a couple comments

File: mod/workspace/cache.js (hunk 7 above)
Comment (Williamas47) [reply mod/workspace/cache.js]: I'm not sure why this is this way, maybe bc it uses on the other two functions too?

Comment (Williamas47) [reply mod/workspace/cache.js]: About sentry, we don't have it configured to this repo at the moment

**Review by fizzy-drinks — APPROVED** [review]

**Review by MatheusAnciloto — APPROVED** [review]
looks good to me! 🚀

**Pull request merged**